File: food-backend/src/models/user.model.ts
import { Schema, model } from "mongoose";
import { Users } from "../types/user.d";

const userSchema = new Schema<Users>(
  {
    username: { type: String, required: true, unique: true, trim: true },
    email: { type: String, required: true, unique: true, trim: true },
    password: { type: String, required: true, minlength: 6 },
    authentication: {
      token: { type: String },
      tokenExpires: { type: Date },
      salt: { type: String },
    },
    resetPasswordToken: { type: String },
    resetPasswordExpires: { type: Date },
  },
  { timestamps: true }
);

export const User = model<Users>("User", userSchema);

File: lib/models/user_model.dart
class User {
  final String id;
  final String username;
  final String email;
  final String? phone;
  

  User({
    required this.id,
    required this.username,
    required this.email,
    this.phone,
    
  });

  factory User.fromJson(Map<String, dynamic> json) => User(
        id: json['_id'],
        username: json['username'],
        email: json['email'],
        phone: json['phone'],
       
      );
}